library/lib.ml

```diff
@@ -419,38 +419,53 @@ type secentry =
     }
   | Context of Univ.ContextSet.t
 
-let sectab =
-  Summary.ref ([] : (secentry list * Opaqueproof.work_list * abstr_list) list)
-    ~name:"section-context"
+type section_data = {
+  sec_entry : secentry list;
+  sec_workl : Opaqueproof.work_list;
+  sec_abstr : abstr_list;
+  sec_poly  : bool;
+}
```

Comment:
A note on style, an alternative is to define a module `SectionData` and have `SectionData.t` plus unqualified record names. This tends to work better for example when passing `poly` to other parts of the system.

Reply:
As section_data is not exported we could also just go with unqualified fields.

Reply:
I agree when the type is exported, but here it's internal, so the actual style doesn't matter much.